modules/objdetect/src/aruco/charuco_detector.cpp

```diff
@@ -362,7 +366,9 @@ void CharucoDetector::setRefineParameters(const RefineParameters& refineParamete
 void CharucoDetector::detectBoard(InputArray image, OutputArray charucoCorners, OutputArray charucoIds,
                                   InputOutputArrayOfArrays markerCorners, InputOutputArray markerIds) const {
     charucoDetectorImpl->detectBoard(image, charucoCorners, charucoIds, markerCorners, markerIds);
-    if (charucoDetectorImpl->checkBoard(markerCorners, markerIds, charucoCorners, charucoIds) == false) {
+    if (charucoDetectorImpl->charucoParameters.checkBoardStructure &&
+        charucoDetectorImpl->checkBoard(markerCorners, markerIds, charucoCorners, charucoIds) == false) {
+        CV_LOG_DEBUG(NULL, "ChArUco board is built incorrectly, you can disable checkBoardStructure flag");
```

Comment:
It's obvious from the function output. Is not it?

Reply:
I tell the user that there is a flag disabling the function checkBoard().